### PR TITLE
feat: port Data.Set.Pairwise

### PR DESCRIPTION
**Pull request by dupuisf**
(Sorry I messed up the branch name)

### CODE REVIEW

File: Mathlib/Data/Set/Pairwise.lean [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+#align pairwise_disjoint_on pairwiseDisjoint_on
+
+theorem PairwiseDisjoint.mono [SemilatticeInf α] [OrderBot α] (hs : Pairwise (Disjoint on f))
+    (h : g ≤ f) : Pairwise (Disjoint on g) :=
+  hs.mono fun i j hij => Disjoint.mono (h i) (h j) hij
+#align pairwise_disjoint.mono PairwiseDisjoint.mono
+
+-- Porting note: The following line causes a panic:
+--alias Function.injective_iff_pairwise_ne ↔ Function.Injective.pairwise_ne _
+
```
Comment (dupuisf) [inline Mathlib/Data/Set/Pairwise.lean:70]: This alias here causes a weird panic, I have no idea what's going on.

File: Mathlib/Data/Set/Pairwise.lean [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+    ((⋃ i ∈ s, f i) \ ⋃ i ∈ t, f i) = ⋃ i ∈ s \ t, f i := by
+  refine'
+    (bunionᵢ_diff_bunionᵢ_subset f s t).antisymm
+      (unionᵢ₂_subset fun i hi a ha => (mem_diff _).2 ⟨mem_bunionᵢ hi.1 ha, _⟩)
+  rw [mem_unionᵢ₂]; rintro ⟨j, hj, haj⟩
+  exact (h (Or.inl hi.1) (Or.inr hj) (ne_of_mem_of_not_mem hj hi.2).symm).le_bot ⟨ha, haj⟩
+#align set.bUnion_diff_bUnion_eq Set.bunionᵢ_diff_bunionᵢ_eq
+
+/-- Equivalence between a disjoint bounded union and a dependent sum. -/
+noncomputable def bUnionᵢEqSigmaOfDisjoint {s : Set ι} {f : ι → Set α} (h : s.PairwiseDisjoint f) :
```
Comment (dupuisf) [inline Mathlib/Data/Set/Pairwise.lean:427]: I wasn't sure how the capitalization rules worked with things like `bUnion`, so I went with this.

**Comment by zeramorphic** [comment]
We have a race condition between two PRs, this and #1175: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Data.2ESet.2EPairwise

**Comment by dupuisf** [comment]
We are going to use #1175, so I'm closing this.

**Pull request closed without merging**